src/coreclr/jit/compiler.h

```diff
@@ -9562,6 +9562,36 @@ class Compiler
         return compIsaSupportedDebugOnly(InstructionSet_AVX512F);
 #else
         return false;
+#endif
+    }
+
+    //------------------------------------------------------------------------
+    // canUseEvexEncodingDebugOnly - Answer the question: Is Evex encoding supported on this target.
+    //
+    // Returns:
+    //    `true` if Evex encoding is supported, `false` if not.
+    //
+    bool canUseEvexEncodingDebugOnly() const
+    {
+#ifdef TARGET_XARCH
+        return (compIsaSupportedDebugOnly(InstructionSet_AVX10v1) || compIsaSupportedDebugOnly(InstructionSet_AVX512F));
+#else
+        return false;
+#endif
+    }
+
+    //------------------------------------------------------------------------
+    // IsAvx10OrIsaSupportedDebugOnly - Answer the question: Is AVX10v1 or the given ISA supported.
+    //
+    // Returns:
+    //    `true` if AVX10v1 or the given ISA is supported, `false` if not.
+    //
+    bool IsAvx10OrIsaSupportedDebugOnly(CORINFO_InstructionSet isa) const
+    {
+#ifdef TARGET_XARCH
+        return (compIsaSupportedDebugOnly(InstructionSet_AVX10v1) || compIsaSupportedDebugOnly(isa));
```

Comment:
Maybe assert that we aren't checking `AVX512F+BW+CD+DQ+VL`, since those are part of the EVEX baseline and could have just been `canUseEvexEncoding()`?

That is, this `IsAvx10OrIsaSupportedDebugOnly` is really only needed for cases like `AVX512+VBMI` where it's not part of the EVEX baseline

Reply:
I have made these changes to make sure we are using ```IsAvx10OrIsaSupportedDebugOnly``` for ```Avx512VBMI``` only.

Reply:
So what I did basically here is that change all the ```IsAvx10OrIsaSupportedDebugOnly``` to ``canUseEvexEncodingDebugOnly```. Hope this is what you meant.

Reply:
That sounds about right, although I think I explained it better in https://github.com/dotnet/runtime/pull/101938#discussion_r1624783410, which is that it boils down to basically:
* `InstructionSet_EVEX` implies `Avx10v1` -or- `Avx512F+CD+DQ+BW+VL` and can be used for 128-bit or 256-bit EVEX encoding instructions in these instruction sets
* `InstructionSet_Avx10v1_V512` should never be queried directly, it is covered by querying `Avx512*`
* `InstructionSet_Avx512F` (and same for `BW`, `CD`, `DQ`) is only queried for 512-bit EVEX encoded instructions
* `InstructionSet_Avx10v1` is only queried for cases like 128-bit/256-bit instructions that wouldn't be in `F+CD+DQ+BW+VL` (such as `VBMI`) and should appear with a corresponding query around `AVX512*_VL` (i.e. `AVX512_VBMI_VL`)

So `IsAvx10OrIsaSupportedDebugOnly` is essentially the debug only check for the 4th bullet point and would be used for something like `Avx512Vbmi_VL`. 

Where-as `Avx512Vbmi` itself would only be queried directly (third bullet point) since its for 512-bit specific usage.

and where `canUseEvexEncodingDebugOnly` represents the first bullet point, which is querying for `Avx10v1` or `Avx512*VL` support, since those are common to the EVEX encoding baseline.

Reply:
Yes. I saw that comment and moved some checks around in my new changes